theseus/embodied/collision/collision.py

```diff
@@ -71,3 +71,8 @@ def _copy_impl(self, new_name: Optional[str] = None) -> "Collision2D":
 
     def dim(self) -> int:
         return 1
+
+    # This is needed so that the SDF container also updates with the new aux var
+    def set_aux_var_at(self, index: int, variable: Variable):
```

Comment:
This fix is okay for now, but maybe worth adding an issue to think about a cleaner version later since it might be a tricky thing for a user to remember to implement something like this for their own custom cost. When inheriting from CostFunction, this is not an obvious thing to do.

Reply:
I agree, but not sure what the best thing would be to do. Having the user register this kind of "special" objects that also require access to variable containers seems a bit awkward as well?

Reply:
Yeah okay, I'll flag an issue. We can think about this more.